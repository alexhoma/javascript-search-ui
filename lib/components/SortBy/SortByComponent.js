"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
exports.__esModule = true;
var preact_1 = require("preact");
var SortByActions_1 = require("./SortByActions");
/**
 * SortBy Filter Component
 */
var SortByComponent = /** @class */ (function (_super) {
    __extends(SortByComponent, _super);
    function SortByComponent() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        /**
         * Handle change
         *
         * @param e
         */
        _this.handleChange = function (e) {
            var props = _this.props;
            var environmentId = props.environmentId;
            var currentQuery = props.currentQuery;
            var repository = props.repository;
            /**
             * Dispatch action
             */
            SortByActions_1.onChangeSearchAction(environmentId, currentQuery, repository, e.target.value);
        };
        return _this;
    }
    /**
     * Should component update
     *
     * @return {boolean}
     */
    SortByComponent.prototype.shouldComponentUpdate = function () {
        return false;
    };
    /**
     * Render
     *
     * @return {any}
     */
    SortByComponent.prototype.render = function (props, state) {
        var containerClassName = props.classNames.container;
        var selectClassName = props.classNames.select;
        var options = props.options;
        var coordinate = props.currentQuery.toArray().coordinate;
        if (!coordinate) {
            options = options.filter(function (o) {
                return o.value != 'distance';
            });
        }
        return (preact_1.h("div", { className: "as-sortBy " + containerClassName },
            preact_1.h("select", { className: "as-sortBy__selector " + selectClassName, onChange: this.handleChange }, options.map(function (option) { return (preact_1.h("option", { value: option.value }, option.name)); }))));
    };
    return SortByComponent;
}(preact_1.Component));
SortByComponent.defaultProps = {
    classNames: {
        container: '',
        select: ''
    }
};
exports["default"] = SortByComponent;
