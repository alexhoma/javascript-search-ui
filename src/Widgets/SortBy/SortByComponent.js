/**
 * @jsx h
 */
import { h, Component } from 'preact';
import {onChangeSearchAction} from "./sortByActions";

/**
 * SortBy Filter Component
 */
class SortByComponent extends Component {
    constructor() {
        super();
    }

    shouldComponentUpdate() {
        return false;
    }

    handleChange = (e) => {
        onChangeSearchAction(
            e.target.value,
            this.props.currentQuery,
            this.props.client
        )
    };

    render() {
        const {
            classNames: {
                container: containerClassName,
                select: selectClassName
            },
            options
        } = this.props;

        return (
            <div className={`asui-sortBy ${containerClassName}`}>
                <select
                    className={`asui-sortBy--selector ${selectClassName}`}
                    onChange={this.handleChange}
                >
                    {options.map(option => (
                        <option value={option.value}>{option.name}</option>
                    ))}
                </select>
            </div>
        );
    }
}

export default SortByComponent;