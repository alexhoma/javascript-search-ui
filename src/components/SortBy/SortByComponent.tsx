import {Component, h} from 'preact';
import {initialSortBySetup, onChangeSearchAction} from "./SortByActions";
import {SortByProps} from './SortByProps';
import {SortByState} from './SortByState';

/**
 * SortBy Filter Component
 */
class SortByComponent extends Component<SortByProps, SortByState> {

    /**
     * Components will mount
     */
    componentWillMount() {

        const props = this.props;
        const environmentId = props.environmentId;
        const options = props.options;
        const currentQuery = props.store.getCurrentQuery();
        const currentOption = options[0].value;

        this.setState({
            value: currentOption,
            visible: false
        });

        /**
         * Dispatch action
         */
        initialSortBySetup(
            environmentId,
            currentQuery,
            currentOption,
        );
    }

    /**
     * Component will receive props
     *
     * @param props
     */
    componentWillReceiveProps(props) {
        this.setState(prevState => {
            return {
                value: props.store.getCurrentQuery().getSortBy().getFirstSortAsString(),
                visible: (props.store.getCurrentResult() != null)
                    ? (props.store.getCurrentResult().getTotalHits() > 0)
                    : false
            }
        })
    }

    /**
     * Handle change
     *
     * @param e
     */
    handleChange = (e) => {

        const props = this.props;
        const environmentId = props.environmentId;
        const currentQuery = props.store.getCurrentQuery();
        const repository = props.repository;
        const currentOption = e.target.value;

        this.setState({
            value: currentOption
        });

        /**
         * Dispatch action
         */
        onChangeSearchAction(
            environmentId,
            currentQuery,
            repository,
            currentOption
        );
    };

    /**
     * Render
     *
     * @return {any}
     */
    render(props, state) {

        const containerClassName = props.classNames.container;
        const selectClassName = props.classNames.select;

        if (!state.visible) {
            return;
        }

        let options = props.options;
        const coordinate = props.store.getCurrentQuery().toArray().coordinate;

        if (!coordinate) {
            options = options.filter(function(o) {
                return o.value != 'distance';
            });
        }

        return (
            <div className={`as-sortBy ${containerClassName}`}>
                <select
                    className={`as-sortBy__selector ${selectClassName}`}
                    onChange={this.handleChange}
                    value={state.value}
                >
                    {options.map(option => (
                        <option value={option.value}>{option.name}</option>
                    ))}
                </select>
            </div>
        );
    }
}

SortByComponent.defaultProps = {
    classNames: {
        container: '',
        select: ''
    }
};

export default SortByComponent;
