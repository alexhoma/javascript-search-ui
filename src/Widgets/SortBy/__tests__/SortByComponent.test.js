/** @jsx h */
import {h} from 'preact';
import {shallow, deep} from 'preact-render-spy';
import SortByComponent from './../SortByComponent';

describe('<SortByComponent />',() => {
    const defaultProps = {
        options: [
            {name: 'Oldest', value: 'year:asc'},
            {name: 'Newest', value: 'year:desc'}
        ]
    }

    it('should render with default props', () => {
        const tree = shallow(<SortByComponent
            target={'#some-tagret'}
            {...defaultProps}
        />);

        expect(tree).toMatchSnapshot();
    });

    it('should render with custom props', () => {
        /**
         * Full custom props available object
         */
        const customProps = {
            options: [
                {name: 'Oldest', value: 'year:asc'},
                {name: 'Newest', value: 'year:desc'}
            ],
            classNames: {
                container: 'custom-container-class'
            }
        }

        const tree = shallow(<SortByComponent
            target={'#some-tagret'}
            {...customProps}
        />);

        expect(tree).toMatchSnapshot();
    });

    it('should handleChange()', () => {
        const tree = shallow(<SortByComponent
            target={'#some-tagret'}
            {...defaultProps}
        />);

        const trigger = jest.fn();
        tree.find('select')[0].attributes.onChange = trigger
        tree.find('select').simulate('change');

        expect(trigger).toBeCalled()
    });
});