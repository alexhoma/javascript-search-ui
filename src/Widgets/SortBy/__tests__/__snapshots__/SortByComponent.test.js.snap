// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SortByComponent /> should render with custom props 1`] = `
preact-render-spy (1 nodes)
-------
<div class="asui-sortBy custom-container-class">
  <select
    class="asui-sortBy--selector undefined"
    onChange={[Function anonymous]}
  >
    <option value="year:asc">Oldest</option>
    <option value="year:desc">Newest</option>
  </select>
</div>

`;

exports[`<SortByComponent /> should render with default props 1`] = `
preact-render-spy (1 nodes)
-------
<div class="asui-sortBy ">
  <select
    class="asui-sortBy--selector "
    onChange={[Function anonymous]}
  >
    <option value="year:asc">Oldest</option>
    <option value="year:desc">Newest</option>
  </select>
</div>

`;
